src/main/java/org/littleshoot/proxy/impl/ProxyConnection.java

```diff
@@ -117,13 +137,25 @@ protected void read(Object msg) {
             readRaw((ByteBuf) msg);
         } else {
             // If not tunneling, then we are always dealing with HttpObjects.
-            readHTTP((HttpObject) msg);
+            try {
+                readHTTP((HttpObject) msg);
+            } catch (RuntimeException ex) {
+                throw wrap(ex);
+            }
         }
     }
 
+    private RuntimeException wrap(RuntimeException ex) {
+        HttpRequest currentHttpRequest = getCurrentHttpRequest();
+        String host = currentHttpRequest.headers().get("Host");
+        return new HttpProxyException(String.format("Host: %s", host), ex);
+    }
```

Comment:
we can't retrieve remoteUser from netty HttpRequest, it means it's impossible to get authenticated user for Forward Proxy